includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -182,6 +184,15 @@ class AMP_Style_Sanitizer extends AMP_Base_Sanitizer {
 	 */
 	private $used_class_names;
 
+	/**
+	 * Regular expression pattern to match focus_class_names in selectors.
+	 *
+	 * The computed pattern is cached to prevent re-constructing for each processed selector.
+	 *
+	 * @var string|null
+	 */
+	private $focus_class_name_selector_pattern;
```

Comment:
I would generally avoid properties for caching purposes only and rather use a static variable within the method that caches. Turning this into a property seems to give it more importance than it actually has, as it seems to be something "actionable".

But that's mostly a matter of preference, this is fine as-is as well.

Reply:
It's a good point, and it's something I debated, but the reason why I went with this approach is that it is _possible_ that the `focus_within_classes` could be changed for the instance of `AMP_Style_Sanitizer` after the initial `sanitize` call. By using a class member variable we can reset it each time the `sanitize` method is called.